skopt/transformers.py

```diff
@@ -0,0 +1,134 @@
+import numpy as np
+from sklearn.preprocessing import LabelBinarizer
+
+# Base class for all 1-D transformers.
+class Transformer(object):
+    def fit(self, X):
+        return self
+
+    def transform(self, X):
+        raise NotImplementedError
+
+    def inverse_transform(self, X):
+        raise NotImplementedError
+
+
+class Identity(Transformer):
+    """Identity transform."""
+
+    def transform(self, X):
+        return X
+
+    def inverse_transform(self, Xt):
+        return Xt
+
+
+class Log10(Transformer):
+    """Base 10 logarithm transform."""
+
+    def transform(self, X):
+        return np.log10(np.asarray(X, dtype=np.float))
+
+    def inverse_transform(self, Xt):
+        return 10.0 ** np.asarray(Xt, dtype=np.float)
+
+
+class CategoricalEncoder(Transformer):
+    """OneHotEncoder that can handle categorical variables."""
+
+    def __init__(self):
+        """Convert labeled categories into one-hot encoded features."""
+        self._lb = LabelBinarizer()
+
+    def fit(self, X):
+        """Fit a list or array of categories.
+
+        Parameters
+        ----------
+        * `X` [array-like, shape=(n_categories,)]:
+            List of categories.
+        """
+        self.mapping_ = {v: i for i, v in enumerate(X)}
+        self.inverse_mapping_ = {i: v for v, i in self.mapping_.items()}
+        self._lb.fit([self.mapping_[v] for v in X])
+        self.n_classes = len(self._lb.classes_)
+
+        return self
+
+    def transform(self, X):
+        """Transform an array of categories to a one-hot encoded representation.
+
+        Parameters
+        ----------
+        * `X` [array-like, shape=(n_samples,)]:
+            List of categories.
+
+        Returns
+        -------
+        * `Xt` [array-like, shape=(n_samples, n_categories)]:
+            The one-hot encoded categories.
+        """
+        return self._lb.transform([self.mapping_[v] for v in X])
+
+    def inverse_transform(self, Xt):
+        """Inverse transform one-hot encoded categories back to their original
+           representation.
+
+        Parameters
+        ----------
+        * `Xt` [array-like, shape=(n_samples, n_categories)]:
+            One-hot encoded categories.
+
+        Returns
+        -------
+        * `X` [array-like, shape=(n_samples,)]:
+            The original categories.
+        """
+        Xt = np.asarray(Xt)
+        return [
+            self.inverse_mapping_[i] for i in self._lb.inverse_transform(Xt)
+        ]
+
+
+class HyperCube(Transformer):
+    """
+    Transform each dimension into [0, 1]
+    """
+    def __init__(self, low, high):
+        self.low = low
+        self.high = high
+
+    def transform(self, X):
```

Comment:
Need some error handling for when the user wants to transform a value outside of the `low -> high` range.